src/AnalyticsEventService.ts

```diff
@@ -0,0 +1,23 @@
+import {EventAPIResponse} from "./EventAPIResponse";
+import {EventPayload} from "./EventPayload";
+
+export interface AnalyticsEventService {
+    /**
+     * Creates a new AnalyticsEventService with the following values defined
+     * as the default when making a report.
+     *
+     * @param payload - desired values to be applied. The new payload will
+     * override any overlapping values.
+     */
+    with(payload: EventPayload): AnalyticsEventService;
+
+    /**
+     * Reports an analytics event. Operand will throw an error if the request
+     * fails. The request will be sent via Beacon API so that requests are
+     * non-blocking.
+     *
+     * @param payload - desired values to be applied. The new payload will
+     * override any overlapping values.
+     */
+    report(payload: object): Promise<EventAPIResponse>;
```

Comment:
I might misremembered this from sprint planning: Do we want EventAPIResponse with or without Promise? @mtian725

Reply:
@alinayh you brought up a good point that I completely overlooked. Beacon doesnt return a promise. It just returns a boolean value but that boolean is whether or not the user agent is added to the request information successfully. It will still throw an error if the request fails, but we have no way to get the response body. 

I discussed this with Ajay last night and we discuss how we might want to go about this because fetch with the keepalive flag is functionally equivalent, but firefox does not support this feature.